snews_pt/snews_pt_utils.py

```diff
@@ -423,3 +427,110 @@ def get_name():
 
     """
     return os.getenv("DETECTOR_NAME")
+
+
+def is_snews_format(snews_message, is_test=False):
+    """ This method checks to see if message meets SNEWS standards.
+
+    Parameters
+    ----------
+    snews_message : dict
+        incoming SNEWS message
+
+    Returns
+    -------
+        bool
+            True if message meets SNEWS standards, else False
+
+    """
+    message_keys = snews_message.keys()
+    missing_key = False
+    contents_bad = False
+    time_bad = False
+
+    log.debug(f"\nChecking message: {snews_message}\n")
+    
+    warning = f'The following Message does not meet SNEWS 2.0 standards!\n{snews_message}\n'
+
+    # Check if detector name is in registered list.
+    detector_file = os.path.abspath(os.path.join(os.path.dirname(__file__), 'auxiliary/detector_properties.json'))
+    with open(detector_file) as file:
+        snews_detectors = json.load(file)
+    snews_detectors = list(snews_detectors.keys())
+
+    if 'detector_name' not in message_keys:
+        warning += f'* Does not have required key: "detector_name"\n'
+        snews_format = False
+        missing_key = True
+
+    elif snews_message['detector_name'] not in snews_detectors:
+        warning += f'* Detector not found: {snews_message["detector_name"]}\n'
+        warning += f"Detector options: {snews_detectors}\n"
+
+    # Check for missing keys
+    if 'neutrino_time' not in message_keys:
+        warning += f'* Does not have required key: "neutrino_time"\n'
+        missing_key = True
+
+    if missing_key:
+        warnings.warn(warning, UserWarning)
+        log.warning(warning)
+        return snews_format
```

Comment:
if `missing_key` is True, it should return the `snews_format` but this can be referenced before assignment. e.g. there is detector_name but no neutrino time, so the `missing_key=True` but there is no `snews_format` defined. <br>
 In the beginning define a `snews_format=True` flag, and after the line 475 set it to False again maybe.

Reply:
Done